blobs/howto/dotnet/dotnet-v12/Concurrency.cs

```diff
@@ -16,79 +17,69 @@ public class Concurrency
+                // This function simulates an external change to the blob after we've fetched it
+                // The external change updates the contents of the blob and the ETag value
+                await SimulateExternalBlobChangesAsync(blobClient);
 
-                using (MemoryStream stream = new MemoryStream(byteArray))
-                {
-                    BlobContentInfo blobContentInfo = await blobClient.UploadAsync(stream, overwrite: true);
-                    originalETag = blobContentInfo.ETag;
-                    Console.WriteLine("Blob added. Original ETag = {0}", originalETag);
-                }
-
-                // This code simulates an update by another client.
-                // No ETag was provided, so original blob is overwritten and ETag updated.
-                string blobContents2 = "Second update overwrites first update.";
-                byteArray = Encoding.ASCII.GetBytes(blobContents2);
+                // Now try to update the blob using the original ETag value
+                string blobContentsUpdate2 = $"{blobContents} Update 2. If-Match condition set to original ETag.";
 
-                using (MemoryStream stream = new MemoryStream(byteArray))
-                {
-                    BlobContentInfo blobContentInfo = await blobClient.UploadAsync(stream, overwrite: true);
-                    Console.WriteLine("Blob updated. Updated ETag = {0}", blobContentInfo.ETag);
-                }
-
-                // Now try to update the blob using the original ETag value.
-                string blobContents3 = "Third update. If-Match condition set to original ETag.";
-                byteArray = Encoding.ASCII.GetBytes(blobContents3);
-
-                // Set the If-Match condition to the original ETag.
-                BlobUploadOptions blobUploadOptions = new BlobUploadOptions()
+                // Set the If-Match condition to the original ETag
+                BlobUploadOptions blobUploadOptions = new()
                 {
                     Conditions = new BlobRequestConditions()
                     {
                         IfMatch = originalETag
                     }
                 };
 
-                using (MemoryStream stream = new MemoryStream(byteArray))
-                {
-                    // This call should fail with error code 412 (Precondition Failed).
-                    BlobContentInfo blobContentInfo = await blobClient.UploadAsync(stream, blobUploadOptions);
-                }
+                // This call should fail with error code 412 (Precondition Failed)
+                BlobContentInfo blobContentInfo = await blobClient.UploadAsync(BinaryData.FromString(blobContentsUpdate2), blobUploadOptions);
             }
-            catch (RequestFailedException e)
+            catch (RequestFailedException e) when (e.Status == (int)HttpStatusCode.PreconditionFailed)
             {
                 if (e.Status == (int)HttpStatusCode.PreconditionFailed)
```

Comment:
I see you've already merged but a quick touchup you can submit separately: you don't need the `if` statement because you've already checked for that condition in the `when` clause. This catch block will only catch if that expression evaluates to true.

Reply:
Ah yes, thank you for mentioning this. I meant to remove that line so thanks for catching it!